test/support/postgresql/import_examples.rb

```diff
@@ -127,3 +127,174 @@ def should_support_postgresql_import_functionality
+          should_update_fields_mentioned
+        end
+
+        context "using string hash map" do
+          let(:update_columns){ { "title" => "title", "author_email_address" => "author_email_address", "parent_id" => "parent_id" } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+
+        context "using string hash map, but specifying column mismatches" do
+          let(:update_columns){ { "title" => "author_email_address", "author_email_address" => "title", "parent_id" => "parent_id" } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned_with_hash_mappings
+        end
+
+        context "using symbol hash map" do
+          let(:update_columns){ { :title => :title, :author_email_address => :author_email_address, :parent_id => :parent_id } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+
+        context "using symbol hash map, but specifying column mismatches" do
+          let(:update_columns){ { :title => :author_email_address, :author_email_address => :title, :parent_id => :parent_id } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned_with_hash_mappings
+        end
+      end
+
+      context "without specifying a unique constraint" do
+        let(:columns){  %w( id title author_name author_email_address parent_id ) }
+        let(:values){ [ [ 100, "Book", "John Doe", "john@doe.com", 17 ] ] }
+        let(:updated_values){ [ [ 100, "Book - 2nd Edition", "This should raise an exception", "john@nogo.com", 57 ] ] }
+
+        macro(:perform_import) do |*opts|
+          Topic.import columns, updated_values, opts.extract_options!.merge(:on_duplicate_key_update => { columns: [:title] }, :validate => false)
+        end
+
+        setup do
+          Topic.import columns, values, :validate => false
+          @topic = Topic.find 100
+        end
+
+        it "should raise an ArgumentError" do
+          assert_raise ArgumentError do
+            perform_import
+          end
+        end
+      end
+
+      context "without specifying update columns" do
+        let(:columns){  %w( id title author_name author_email_address parent_id ) }
+        let(:values){ [ [ 100, "Book", "John Doe", "john@doe.com", 17 ] ] }
+        let(:updated_values){ [ [ 100, "Book - 2nd Edition", "This should raise an exception", "john@nogo.com", 57 ] ] }
+
+        macro(:perform_import) do |*opts|
+          Topic.import columns, updated_values, opts.extract_options!.merge(:on_duplicate_key_update => { key: :id }, :validate => false)
+        end
+
+        setup do
+          Topic.import columns, values, :validate => false
+          @topic = Topic.find 100
+        end
+
+        it "should raise an ArgumentError" do
+          assert_raise ArgumentError do
+            perform_import
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
@jkowens, do you think we can make the default behavior and interface for PostgreSQL compatible with MySQL? If so, I'd love to move all of the baseline examples into a single spec and then leave any examples that are MySQL or PostgreSQL specific (incompatible APIs) into their respective specs. 

Thoughts?